### PR TITLE
Fixes players going through each other - #992

### LINKED ISSUE #992: Players aren't solid anymore (can walk through each other) B: 30
# Preliminary information

## Category
BUG

## Bug: Release or Development?
Develop

# Report
## Current Behavior
In title.
Looks like it's a regression caused by matrix switch functionality and/or changes to colliders

## Expected/Wanted/Requested Behavior
Players should not be able to pass freely through each other (until we have intents in, at least)

## Steps to Reproduce
Please enter the steps to reproduce the bug or behaviour:
1. Get two players on the server
2. Try walking through each other

## Network
All types

### PR DESCRIPTION
**Pull request by badbon**
### Purpose
Fixes #992

### Approach
I went through a bit of hell because I am still new to codebase, but I believe I've found a fix. 👍 Basically the past code wasn't even checking for players, it was just checking for tiles (I guess?) I made a specific checks for a registered player "tile", but then issue came up with network replays which I also had to fix by checking if that player tile is same player that called the function for player check, because player was stopping moving thinking it hit a player, but it was actually itself, just being called from network replay. (I hope this makes sense lol)

### Open Questions and Pre-Merge TODOs

- [x]  I read the contribution guidelines and the wiki.
- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  This PR does not include files without specific need to do so.
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  This PR has been tested in multiplayer

### Notes:
Going to PM my PayPal to Doobly in case this gets merged.

### CODE REVIEW

File: UnityProject/Assets/Scripts/PlayGroups/PlayerMove.cs [hunk 1]
```diff
@@ -200,49 +200,62 @@ private Vector3Int GetMoveDirection(KeyCode action)
...
-			if (playerSync.PullingObject != null) {
-				if (curMatrix.ContainsAt(newPos, playerSync.PullingObject)) {
-					return direction;
-				}
-			}
-
-			//could not pass
-			return Vector3Int.zero;
-		}
+                if (isGhost)
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/PlayGroups/PlayerMove.cs]: please use tabs for indentation instead of spaces!

**Comment by badbon** [comment]
Not sure if I messed it up worse or fixed it, try now :c

**Comment by DooblyNoobly** [comment]
@NicholasM10 I'm having trouble seeing where the Player check is in PlayerMove. Can you direct me to the line # of the changes (hard to tell with the indent changes)

**Comment by DooblyNoobly** [comment]
Oh right you added it to the ContainsAt on Matrix

**Comment by DooblyNoobly** [comment]
I tested in high lag mp. Works well. We may need a new issue for Machines and Objects

**Comment by DooblyNoobly** [comment]
Also I see issues in the future when intent comes into play. But we will cross that bridge when we get there

**Pull request merged**